mlflow/diviner.py

```diff
@@ -208,47 +288,66 @@ def load_model(model_uri, dst_path=None):
                       `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
                       artifact-locations>`_.
     :param dst_path: The local filesystem path to which to download the model artifact.
-                     This directory must already exist. If unspecified, a local output
+                     This directory must already exist if provided. If unspecified, a local output
                      path will be created.
 
     :return: A ``Diviner`` model instance
     """
+    model_uri = str(model_uri)
+    # Determine if the file read indicator is present for saved in Spark or not
+    if RunsArtifactRepository.is_runs_uri(model_uri):
+        runs_uri = model_uri
+        model_uri = RunsArtifactRepository.get_underlying_uri(model_uri)
+        _logger.info("'%s' resolved as '%s'", runs_uri, model_uri)
+    elif ModelsArtifactRepository.is_models_uri(model_uri):
+        runs_uri = model_uri
+        model_uri = ModelsArtifactRepository.get_underlying_uri(model_uri)
+        _logger.info("'%s' resolved as '%s'", runs_uri, model_uri)
```

Comment:
While we're here, can we add this code to `_get_flavor_configuration_from_uri` instead and also remove it from `mlflow/spark.py`?

Reply:
refactored.